src/trust-region/basic-trust-region.jl

```diff
@@ -0,0 +1,40 @@
+export TrustRegion
+
+"""Basic trust region type."""
+type TrustRegion <: AbstractTrustRegion
+  initial_radius :: Float64
+  radius :: Float64
+  max_radius :: Float64
+  acceptance_threshold :: Float64
+  increase_threshold :: Float64
+  decrease_factor :: Float64
+  increase_factor :: Float64
+  ratio :: Float64
```

Comment:
I made this change because TRON updates the trust region using the ratio. The alternative would be changing `update!` to also receive `ratio`, and change `trunk` accordingly. Another alternative would be not having unified trust region `update!`.